cmake/TryAppendLinkerFlags.cmake

```diff
@@ -0,0 +1,32 @@
+include_guard(GLOBAL)
+
+include(CheckCSourceCompiles)
+
+function(secp256k1_check_linker_flags_internal flags output)
+  string(MAKE_C_IDENTIFIER "${flags}" result)
+  string(TOUPPER "${result}" result)
+  set(result "LINKER_SUPPORTS_${result}")
+  set(CMAKE_REQUIRED_FLAGS "${flags}")
+  if(MSVC)
+    string(APPEND " /WX")
+  elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+    string(APPEND " -Wl,-fatal_warnings")
+  else()
+    string(APPEND " -Wl,--fatal-warnings")
```

Comment:
Shouldn't this depend on the linker and not on the OS?

Reply:
I don't think it can be implemented in a simple way; see https://gitlab.kitware.com/cmake/cmake/-/issues/18209.

However, we can just check each variant.

Reply:
Okay, I think checking `CMAKE_SYSTEM_NAME` is good enough and pragmatic. GNU ld doesn't even support the creation of macOS binaries, and even clang's macOS linker `ld64.lld` uses `-fatal_warnings`...

But shouldn't we append `-Wl,-fatal_warnings` or `-Wl,--fatal-warnings` also for the autotools build?